File: D2.MasterData/Models/BoundSubUnit.cs
using D2.MasterData.Parameters;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace D2.MasterData.Models
{
    public class BoundSubUnit: SubUnit
    {
        protected BoundSubUnit() { }

        public BoundSubUnit(BoundSubUnitParameters argument, Entrance entrance, AdministrationUnit administrationUnit)
        {
            Floor = argument.Floor;
            Id = argument.Id;
            Title = argument.Title;
            Number = argument.Number;
            Usage = argument.Usage;
            Version = argument.Version;
            Entrance = entrance;
            AdministrationUnit = administrationUnit;
        }

        public virtual string Floor
        {
            get;
            protected set;
        }

        public virtual Entrance Entrance
        {
            get;
            protected set;
        }
    }
}

File: D2.UI/D2.UI/src/app/masterdata/subunit/boundsubunit.ts
import {SubUnit} from './subunit';

export class BoundSubunit extends SubUnit {
  public Floor?; string;
  constructor (id: string, title: string, version: number, usage: string, number: number, floor: string) {
    super(id, title, version, number);
    this.Floor = floor;
  }
}
